monai/transforms/spatial/array.py

```diff
@@ -2609,3 +2619,168 @@ def _get_params(
+
+    Args:
+        patch_size: size of patches to generate slices for, 0 or None selects whole dimension
+        start_pos: starting position in the array, default is 0 for each dimension.
+            np.random.randint(0, patch_size, 2) creates random start between 0 and `patch_size` for a 2D image.
+        fix_num_patches: number of patches to return. Defaults to None, which returns all the available patches.
+        overlap: the amount of overlap of neighboring patches in each dimension (a value between 0.0 and 1.0).
+            If only one float number is given, it will be applied to all dimensions. Defaults to 0.0.
+        sort_key: a callable or string that defines the order of the patches to be returned. If it is a callable, it
+            will be passed directly to the `key` argument of `sorted` function. The string can be "min" or "max",
+            which are, respectively, the minimum and maximum of the sum of intensities of a patch across all dimensions
+            and channels. Also "random" creates a random order of patches.
+            By default no sorting is being done and patches are returned in a row-major order.
+        pad_mode: {``"constant"``, ``"edge"``, ``"linear_ramp"``, ``"maximum"``, ``"mean"``,
+            ``"median"``, ``"minimum"``, ``"reflect"``, ``"symmetric"``, ``"wrap"``, ``"empty"``}
+            One of the listed string values or a user supplied function. Defaults to ``"wrap"``.
+            See also: https://numpy.org/doc/stable/reference/generated/numpy.pad.html
+        pad_opts: padding options, see `numpy.pad`
+
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        patch_size: Sequence[int],
+        start_pos: Sequence[int] = (),
+        fix_num_patches: Optional[int] = None,
+        overlap: Union[Sequence[float], float] = 0.0,
+        sort_key: Optional[Union[Callable, str]] = None,
+        pad_mode: Union[NumpyPadMode, str] = NumpyPadMode.CONSTANT,
+        pad_opts: Optional[Dict] = None,
+    ):
+        self.patch_size = ensure_tuple(patch_size)
+        self.start_pos = ensure_tuple(start_pos)
+        self.pad_mode: NumpyPadMode = look_up_option(pad_mode, NumpyPadMode)
+        self.pad_opts = {} if pad_opts is None else pad_opts
+        self.overlap = overlap
+        self.num_patches = 0
+        self.fix_num_patches = fix_num_patches
+        self.sort_key: Optional[Callable]
+        if isinstance(sort_key, str):
+            if sort_key == "random":
+                self.sort_key = np.random.random
+            if sort_key == "min":
+                self.sort_key = self._get_patch_sum
+            if sort_key == "max":
+                self.sort_key = self._get_negative_patch_sum
+            else:
+                ValueError(f'sort_key should be either "min", "max", or "random", "{sort_key}" was given.')
+        else:
+            self.sort_key = sort_key
+
+    @staticmethod
```

Comment:
It's a staticmethod but starts with "_", I feel confused it's for public use or not?

Thanks.

Reply:
I removed "_".